### PR TITLE
[SPARK-21902][CORE] Uniform calling for DiskBlockManager.getFile

### PR DESCRIPTION
**Pull request by caneGuy**
## What changes were proposed in this pull request?

Originally this pr is for [SPARK-21902](https://issues.apache.org/jira/browse/SPARK-21902) which i think calling for DiskBlockManager.getFile should be uniform and we should print root cause when call 'Blockmanager#doPut' failed.
Since @kiszk commented,i split this one into an other pr [PR-19171](https://github.com/apache/spark/pull/19171)

## How was this patch tested?
Exsist unit test

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/storage/DiskBlockManager.scala [hunk 1]
```diff
@@ -53,7 +53,7 @@ private[spark] class DiskBlockManager(conf: SparkConf, deleteFilesOnStop: Boolea
   /** Looks up a file by hashing it into one of our local subdirectories. */
   // This method should be kept in sync with
   // org.apache.spark.network.shuffle.ExternalShuffleBlockResolver#getFile().
-  def getFile(filename: String): File = {
+  private def getFileInternal(filename: String): File = {
```
Comment (kiszk) [inline core/src/main/scala/org/apache/spark/storage/DiskBlockManager.scala]: Why do we need this rename?

**Comment by kiszk** [comment]
Since this PR includes two types of changes, I think that it would be good to split this into two PRs if both of them are necessary.

**Comment by caneGuy** [comment]
Ping @kiszk Cloud you help take a look at this? Thanks too much.

**Comment by jerryshao** [comment]
This is not a necessary fix. We usually don't do such changes without really fix anything.

**Comment by caneGuy** [comment]
Actually , initially i put this together with  [PR-19171](https://github.com/apache/spark/pull/19171) since i found the api is not unify when fix that problem.All right i will close this one.Cloud you help review 19171? @jerryshao Thanks.

**Pull request closed without merging**